builtin/stash--helper.c

```diff
@@ -634,6 +640,55 @@ static int list_stash(int argc, const char **argv, const char *prefix)
 	return ret;
 }
 
+static int show_stash(int argc, const char **argv, const char *prefix)
+{
+	int i, ret = 0;
+	struct argv_array args = ARGV_ARRAY_INIT;
+	struct argv_array args_refs = ARGV_ARRAY_INIT;
+	struct stash_info info;
+	struct option options[] = {
+		OPT_END()
+	};
+
+	argc = parse_options(argc, argv, prefix, options,
+			git_stash_helper_show_usage, PARSE_OPT_KEEP_UNKNOWN);
+
+	argv_array_push(&args, "diff");
+
+	for (i = 0; i < argc; ++i) {
+		if (argv[i][0] == '-')
+			argv_array_push(&args, argv[i]);
+		else
+			argv_array_push(&args_refs, argv[i]);
+	}
+
+	ret = get_stash_info(&info, args_refs.argc, args_refs.argv);
+	argv_array_clear(&args_refs);
+
+	if (!ret && args.argc == 1) {
+		int showStat = 1, showPatch = 0;
+
+		git_config_get_bool("stash.showStat", &showStat);
+		if (showStat)
+			argv_array_push(&args, "--stat");
+
+		git_config_get_bool("stash.showPatch", &showPatch);
+		if (showPatch)
+			argv_array_push(&args, "-p");
```

Comment:
This is a pretty verbatim copy of the Unix shell code. In C, however, we can do better: by calling `git_config(<fn>)` where `<fn>` is a callback function that consumes the parsed config values. Please have a look at `git grep -w git_config -- builtin/`'s output for inspiration.

Reply:
I guess that this would be the best way to proceed. In the future, there might be added other config options.

Reply:
I done this while refactoring `show_stash()` to use the diff API, as you suggested [here](https://github.com/git/git/pull/495#discussion_r191661468).